orbstation/modules/cargo/cadavers.dm

```diff
@@ -0,0 +1,128 @@
+	for (var/part in H.internal_organs)
+		if(istype(part, /obj/item/organ/internal/heart))
+			qdel(part)
+
+/datum/supply_pack/medical/cadaver/felinid
+	name = "Medical Cadaver Crate (Felinid)"
+	desc = "A preserved medical cadaver, for dissections, spare organs, or experiments. This one is a felinid."
+	contains = list(/mob/living/carbon/human/species/felinid)
+
+/datum/supply_pack/medical/cadaver/human
+	name = "Medical Cadaver Crate (Human)"
+	desc = "A preserved medical cadaver, for dissections, spare organs, or experiments. This one is a human."
+	contains = list(/mob/living/carbon/human)
+
+/datum/supply_pack/medical/cadaver/jelly
+	name = "Medical Cadaver Crate (Jellyperson)"
+	desc = "A preserved medical cadaver, for dissections, spare organs, or experiments. This one is a jellyperson."
+	contains = list(/mob/living/carbon/human/species/jelly)
+
+/datum/supply_pack/medical/cadaver/moth
+	name = "Medical Cadaver Crate (Moth)"
+	desc = "A preserved medical cadaver, for dissections, spare organs, or experiments. This one is a moth."
+	contains = list(/mob/living/carbon/human/species/moth)
+
+/datum/supply_pack/medical/cadaver/pod
+	name = "Medical Cadaver Crate (Podperson)"
+	desc = "A preserved medical cadaver, for dissections, spare organs, or experiments. This one is a podperson."
+	contains = list(/mob/living/carbon/human/species/pod)
+
+/datum/supply_pack/medical/cadaver/rat
+	name = "Medical Cadaver Crate (Ratfolk)"
+	desc = "A preserved medical cadaver, for dissections, spare organs, or experiments. This one is a ratfolk."
+	contains = list(/mob/living/carbon/human/species/ratfolk)
+
+/datum/supply_pack/medical/cadaver/tiziran
+	name = "Medical Cadaver Crate (Tiziran)"
+	desc = "A preserved medical cadaver, for dissections, spare organs, or experiments. This one is a tiziran."
+	contains = list(/mob/living/carbon/human/species/lizard)
+
+// I'm making this a subtype of coffin solely so vampires can sleep in them.
+/obj/structure/closet/crate/coffin/stasis
+	name = "stasis coffin"
+	desc = "An advanced coffin used to keep corpses fresh for extended transport. Once opened, it will permanently stop functioning."
+	icon = 'orbstation/icons/obj/coffin.dmi'
+	icon_state = "stasiscoffin"
+	resistance_flags = NONE
+	max_integrity = 70
+	material_drop = /obj/item/stack/sheet/iron
+	material_drop_amount = 5
+	open_sound = 'sound/machines/closet_open.ogg'
+	close_sound = 'sound/machines/closet_close.ogg'
+	open_sound_volume = 25
+	close_sound_volume = 50
+	can_install_electronics = FALSE
+	mob_storage_capacity = 1 //no mass stasis, sorry
+	var/functional = TRUE //whether the coffin still works or not
+
+/obj/structure/closet/crate/coffin/stasis/closet_update_overlays(list/new_overlays)
+	. = new_overlays
+	if(!opened && functional)
+		. += "stasisactive"
+	if(manifest)
+		. += "manifest"
+
+/obj/structure/closet/crate/coffin/stasis/open(mob/living/user, force = FALSE)
+	if(functional)
+		functional = FALSE
+		for(var/atom/movable/AM in src)
```

Comment:
Actually this should probably just be
`for (var/mob/living/carbon/corpse in src)`
Loops can do type checking themselves.